message ByteArrayHeader {
	required string datasource = 1;
	required string filename = 2;
	required int64 offset = 3;
	required int64 size = 4;
}